simplipy/api.py

```diff
@@ -237,6 +251,38 @@ async def _async_api_request(
 
         return data
 
+    async def async_media(self, url: str) -> bytes | None:
```

Comment:
If you change line 91 to `self.async_media = self._wrap_request_method(` (similar to what occurs on line 86), can we get rid of this extra method? You'll probably need to do some typing work, but that is preferable to having another method defined.

Reply:
Believe me, I tried.  _wrap_request_method() is designed to return JSON (dict[str, Any]) from api calls to the normal SS rest api.  I tried changing it so it could return a dict, or bytes or None, but:

1.  I could not, even using Union
2. It doesn't seem "cool" for it to be so flexible

The public call to get api results **should** return JSON, and the public call to get media files **should** return bytes.  I think my current implementation meets the constraints imposed from trying to have a good public interface, while trying to reuse code (auth, retries) and satisfy strict type checking.